watch/inotify.go

```diff
@@ -134,3 +141,27 @@ func (fw *InotifyFileWatcher) ChangeEvents(t *tomb.Tomb, pos int64) (*FileChange
 
 	return changes, nil
 }
+
+func (fw *InotifyFileWatcher) checkAndNotifyIfModifiedInBetween(changes *FileChanges) (os.FileInfo, bool) {
+	fi, err := os.Stat(fw.Filename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			RemoveWatch(fw.Filename)
+			changes.NotifyDeleted()
+			return nil, true
+		}
+		util.Fatal("Failed to stat file %v: %v", fw.Filename, err)
+	}
+
+	if fw.Size > 0 && fw.Size > fi.Size() { // old file size was larger than now => truncated
```

Comment:
Can `fi.Size()` ever be negative? If not, `fw.Size() > 0` is redundant

Reply:
I can't imagine os.Stat would return a negative file size. Removed